lib/agent/providers/hardware/windows.js

```diff
@@ -77,75 +77,81 @@ var firmware_keys = {
 
 exports.get_firmware_info = function(callback) {
 
-  if (gte(common.os_release,"10.0.0")) { //10.0.0 is w10
-
-    let data = {"uuid":"","serial_number":"","bios_vendor":"","bios_version":"","mb_vendor":"","mb_serial":"","mb_model":"",
-    "mb_version":"","device_type":""};
-
-     si.system((stdoutsi) => {
-      if (!stdoutsi || !stdoutsi.uuid || stdoutsi.uuid.toString().trim() == ''){
-        callback(new Error('No Info found.'));
-      }
-      else {
-        data.uuid = stdoutsi.uuid;
-        data.serial_number = stdoutsi.serial;
-
-        si.bios((stdoutsi) => {
-          if (!stdoutsi || !stdoutsi.vendor || stdoutsi.vendor.toString().trim() == ''){
-            callback(null,data)
+    var count = 0, data = {};
+    var fetch = function(key, section, value){
+      wmic.get_value(section, value, null, function(err, res){
+
+        if (key == 'device_type'){
+          res = err ? 'Desktop' : 'Laptop'
+          data[key] = res;
+        }
+        if (!err && res)
+          data[key] = res;
+          logger.info("hardware info3 : "+ JSON.stringify(data))
```

Comment:
It's better to take out this logger here.

Reply:
done